pkg/odo/genericclioptions/runnable.go

```diff
@@ -4,20 +4,60 @@ import (
 	"flag"
 	"fmt"
 	"os"
+	"strconv"
+	"time"
+
+	"github.com/openshift/odo/pkg/odo/cli/ui"
+	"gopkg.in/AlecAivazis/survey.v1"
+
+	"github.com/openshift/odo/pkg/preference"
+	"github.com/openshift/odo/pkg/segment"
+	"k8s.io/klog"
 
 	"github.com/openshift/odo/pkg/log"
 	"github.com/openshift/odo/pkg/odo/util"
 	"github.com/spf13/cobra"
 	"github.com/spf13/pflag"
 )
 
+var (
+	segmentClient *segment.Client
+)
+
 type Runnable interface {
 	Complete(name string, cmd *cobra.Command, args []string) error
 	Validate() error
 	Run() error
 }
 
 func GenericRun(o Runnable, cmd *cobra.Command, args []string) {
+	var err error
+	var startTime time.Time
+	cfg, _ := preference.New()
+
+	// Prompt the user to consent for telemetry if a value is not set already
+	// Skip prompting if the preference command is called
+	// This prompt has been placed here so that it does not prompt the user when they call --help
```

Comment:
Great work here! It's not prompting for any `-h` commands - `odo -h`, `odo service -h`, etc.

Reply:
I think @kadel & @girishramnani had different views than mine.